lib/prm/doctor_api/api.ex

```diff
@@ -38,17 +38,123 @@ defmodule PRM.DoctorAPI do
     fields = ~W(
       mpi_id
       status
-      education
-      certificates
-      licenses
       jobs
       active
       name
       created_by
       updated_by
     )
 
+    available_statuses = [
+      "PENDING_APPROVAL",
+      "APPROVED",
+      "DECLINED"
+    ]
+
     doctor
     |> cast(attrs, fields)
+    |> cast_embed(:education, with: &education_changeset/2)
+    |> cast_embed(:certificates, with: &certificate_changeset/2)
+    |> cast_embed(:licenses, with: &medical_license_changeset/2)
+    |> validate_inclusion(:status, available_statuses)
+  end
+
+  defp education_changeset(education, attrs) do
+    education_fields = ~W(
+      institution_name
+      certificate_number
+      degree
+      start_date
+      finished_date
+      speciality
+    )
+
+    required_fields = [
+      :institution_name,
+      :certificate_number,
+      :degree,
+      :start_date,
+      :finished_date,
+      :speciality
+    ]
+
+    available_degrees = [
+      "Doctor of Medicine",
+      "Bachelor of Medicine",
+      "Bachelor of Surgery",
+      "Doctor of Osteopathic Medicine",
+      "Master of Clinical Medicine"
+    ]
+
+    available_specialities = [
+      "Загальна практика - сімейна медицина",
+      "Педіатрія",
+      "Підліткова терапія",
+      "Терапія"
```

Comment:
I'd love to but this question has a bigger scope.

We need to define & stay in sync with what comes from an upstream service, be it Integration Layer or UI.